addons/l10n_ro_edi/models/account_edi_xml_ubl_ciusro.py

```diff
@@ -76,6 +82,26 @@ def _export_invoice_vals(self, invoice):
             'tax_currency_code': 'RON',
         })
 
+        # Deal with legal_entity_vals here, as there is no way to distinguish between customer and supplier in _get_partner_party_legal_entity_vals_list
+        for legal_entity_vals in vals['vals']['accounting_supplier_party_vals']['party_vals']['party_legal_entity_vals']:
+            supplier = legal_entity_vals['commercial_partner']
+            if (
+                (
+                    not supplier.vat
+                    or len(supplier.vat) == 1
+                )
+                and supplier.company_registry
+            ):
+                legal_entity_vals['company_id'] = supplier.company_registry
+
+        for legal_entity_vals in vals['vals']['accounting_customer_party_vals']['party_vals']['party_legal_entity_vals']:
+            customer = legal_entity_vals['commercial_partner']
+            if (
+                not customer.vat
+                or len(customer.vat) == 1
+            ):
+                legal_entity_vals['company_id'] = DEFAULT_VAT
```

Comment:
```suggestion
        for type in ('supplier', 'customer'):
            for legal_entity_vals in vals['vals'][f'accounting_{type}_party_vals']['party_vals']['party_legal_entity_vals']:
                partner = legal_entity_vals['commercial_partner']
                company_registry = partner.company_registry if type == 'supplier' else True
                company_id = partner.company_registry if type == 'supplier' else DEFAULT_VAT
                if (
                    (not partner.vat or len(partner.vat) == 1)
                    and company_registry
                ):
                    legal_entity_vals['company_id'] = company_id
```
We could also do something like that. I'm not sure it's better tho haha.